instance/models/openedx_instance.py

```diff
@@ -117,6 +117,10 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         for appserver in active_appservers:
             server_name = "appserver-{}".format(appserver.pk)
             ip_address = appserver.server.public_ip
+            if not ip_address:
+                # if the ip_address is not there, try to update_status() before failing.
```

Comment:
This comment is a code smell: there is little that we gain from it which the code doesn't already make clear right after.

Also, by *fail*, we actually literally mean to make this function fail, i.e. cancel the load balancer reconfiguration process (This probably sounds confusing, so see https://github.com/open-craft/ansible-load-balancer#architecture-notes if it may be helpful), which probably entails throwing an exception or something to prevent Ocim from actually reconfiguring the load balancer in this state.

Reply:
(BTW, for the comment itself, we can probably still have a comment but just rewrite it to explain in some more detail the motive behind this -- up to you).